addons/point_of_sale/static/src/js/Screens/ClientListScreen/ClientListScreen.js

```diff
@@ -124,20 +127,39 @@ odoo.define('point_of_sale.ClientListScreen', function(require) {
             this.state.detailIsShown = true;
             this.render();
         }
-        clickNext() {
+        async clickNext() {
+            if (this.isUnsavedChanges() && await this.isSaveWanted()) {
+                await this.saveChanges();
+            }
             this.state.selectedClient = this.nextButton.command === 'set' ? this.state.selectedClient : null;
             this.confirm();
         }
+        isUnsavedChanges() {
+            return Object.keys(this.changes).length > 0  ? true: false;
```

Comment:
this line can be simply be: `return Object.keys(this.changes).length > 0`

Reply:
good one!! thanks!